include/deal.II/lac/sparse_matrix_ez.h

```diff
@@ -683,6 +683,17 @@ class SparseMatrixEZ : public Subscriptor
   const_iterator begin (const size_type r) const;
 
   /**
+   * Iterator on diagonal entry of row @p r.
+   *
+   * @warning If the current row has no diagonal element, then the returned
+   * iterator will be same as end(r), i.e., an iterator points to first element
+   * of next row or same place behind end of the entire matrix.
+   * So, do not dereference or query information to the returned iterator blindly
+   * unless you can make sure your matrix do have a diagonal element in row @p r.
```

Comment:
This seems like a poor choice of returned value, because it typically points to a valid entry -- it's far too simply for people to create a bug in their code by not checking that the element they got is what they had intended. How about just returning `end()` in that case?